src/Microsoft.Extensions.Localization.Abstractions/IPluralizeStringLocalizer.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved. 
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+namespace Microsoft.Extensions.Localization
+{
+    /// <summary>
+    /// Represents a service that provides localized strings with pluralization support.
+    /// </summary>
+    public interface IPluralizeStringLocalizer : IStringLocalizer
+    {
+        /// <summary>
+        /// Gets the plural form for a given <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The name of the string resource.</param>
+        /// <param name="count">The number of items you are trying to pluralize.</param>
+        /// <returns>The pluralized string resource as a <see cref="LocalizedString"/>.</returns>
+        LocalizedString Pluralize(string name, int count);
+
+        /// <summary>
+        /// Gets the pluralization rule for a given language.
+        /// </summary>
+        /// <param name="twoLetterISOLanguageName">The language that used to retrieve the plural forms for.</param>
+        /// <returns> The pluralization rule as <see cref="PluralizationRule"/>.</returns>
+        PluralizationRule GetPluralRule(string twoLetterISOLanguageName);
```

Comment:
May be, there 're different ways to think about this, either we pass the number that decide which form should be selected, but what i'm looking for is a simple and elegant way that code reduce the efforts by passing the language rule once and the APIs will take the rest of the stuff to find the proper form, so that's why I provide a code snippet in the unit tests to prove the concept.
Also we may think about better option to provide the language pluralization rules without tedious and repetitive code, AFAIK PO files are one of the format that support the language rule itself as a part of the file itself to choose the appropriate form. Hope we hear the feedback from other folks